src/transform/mod.rs

```diff
@@ -276,3 +276,88 @@ pub fn inverse_transform_add(
     _ => panic!("unimplemented tx size")
   }
 }
+
+#[cfg(test)]
+mod test {
+  use super::*;
+  use rand::random;
+
+  fn test_roundtrip(tx_size: TxSize, tx_type: TxType) {
+    let mut src_storage = [0u16; 64 * 64];
+    let src = &mut src_storage[..tx_size.area()];
+    let mut dst_storage = [0u16; 64 * 64];
+    let dst = &mut dst_storage[..tx_size.area()];
+    let mut res_storage = [0i16; 64 * 64];
+    let res = &mut res_storage[..tx_size.area()];
+    let mut freq_storage = [0i32; 64 * 64];
+    let freq = &mut freq_storage[..tx_size.area()];
+    for ((r, s), d) in res.iter_mut().zip(src.iter_mut()).zip(dst.iter_mut()) {
+      *s = random::<u8>() as u16;
+      *d = random::<u8>() as u16;
+      *r = (*s as i16) - (*d as i16);
+    }
+    forward_transform(res, freq, tx_size.width(), tx_size, tx_type, 8);
+    inverse_transform_add(freq, dst, tx_size.width(), tx_size, tx_type, 8);
+
+    for (s, d) in src.iter().zip(dst) {
+      assert!(i16::abs((*s as i16) - (*d as i16)) <= 1);
+    }
+  }
+
+  #[test]
+  fn roundtrips() {
+    use partition::TxSize::*;
+    use partition::TxType::*;
+    let combinations = vec![
```

Comment:
This can just be a static array.

Reply:
I had tried that but had problem with IntoIter trait not implemented for arrays larger than 32. I looked some more now and it seems that the iter method works for all sizes. I will push this change soon.